File: DatingPortal/src/app/models/message.ts
export class Message {
  id: number;

  senderId: string;
  senderKnownAs: string;
  senderPhotoUrl: string;

  recipientId: string;
  recipientKnownAs: string;
  recipientPhotoUrl: string;

  sentDate: Date;
  readDate: Date;

  isRead: boolean;
  isSenderDeleted: boolean;
  isRecipientDeleted: boolean;

  content: string;
}

File: DatingApi/Data/Models/Message.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace DatingApi.Data.Models
{
    public class Message
    {
        public int Id { get; set; }
    
        public string SenderId { get; set; }
        public User Sender { get; set; }

        public string RecipientId { get; set; }
        public User Recipient { get; set; }

        public DateTime SentDate { get; set; }
        public DateTime ReadDate { get; set; }

        public bool IsRead { get; set; }
        public bool IsSenderDeleted { get; set; }
        public bool IsRecipientDeleted { get; set; }

        public string Content { get; set; }
    }
}